File: backend/Services/Implementations/VendorService.cs
using backend.Models.Entities;
using backend.Repositories;

namespace backend.Services
{
    public class VendorService : IVendorService
    {
        private readonly IVendorRepository _vendorRepository;

        public VendorService(IVendorRepository vendorRepository)
        {
            _vendorRepository = vendorRepository;
        }

        public async Task<List<Vendor>> GetAllVendors()
        {
            return await _vendorRepository.GetAllAsync();
        }

        public async Task<Vendor?> GetVendorById(long id)
        {
            return await _vendorRepository.GetByIdAsync(id);
        }

        public async Task<List<Vendor>> GetVendorsByCategory(string category)
        {
            return await _vendorRepository.GetByCategoryAsync(category);
        }

        public async Task<List<Vendor>> GetPremiumVendors()
        {
            return await _vendorRepository.GetPremiumVendorsAsync();
        }

        public async Task<Vendor> CreateVendor(Vendor vendor)
        {
            await _vendorRepository.AddAsync(vendor);
            return vendor;
        }
    }
}

File: springboot-backend/src/main/java/com/wedspot/backend/services/implementation/VendorService.java
package com.wedspot.backend.services.implementation;

import com.wedspot.backend.Model.Entity.Vendor;
import com.wedspot.backend.repository.IVendorRepository;
import com.wedspot.backend.services.IVendorService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class VendorService implements IVendorService {

    private final IVendorRepository vendorRepository;

    @Override
    public List<Vendor> getAllVendors() {
        return vendorRepository.findAll();
    }

    @Override
    public Optional<Vendor> getVendorById(Long id) {
        return vendorRepository.findById(id);
    }

    @Override
    public List<Vendor> getVendorsByCategory(String category) {
        return vendorRepository.findByCategory(category);
    }

    @Override
    public List<Vendor> getPremiumVendors() {
        return vendorRepository.findByPremiumTrue();
    }

    @Override
    public Vendor createVendor(Vendor vendor) {
        return vendorRepository.save(vendor);
    }
}
